py_wtf/repository.py

```diff
@@ -34,28 +40,36 @@ def _load_from_disk(self, key: ProjectName) -> None:
         index_file = self._index_file(key)
         index_contents = index_file.read_bytes()
         proj = converter.loads(index_contents, Project)
-        self._cache[key] = proj
+        loop = asyncio.get_event_loop_policy().get_event_loop()
+        self._cache[key] = Future(loop=loop)
+        self._cache[key].set_result(proj)
 
```

Comment:
I don't understand this part. We create a new future, and immediately set its result? (Same in `_save`). If we have the result, then why create a `Future` at all? If there was a `Future` in there already, then anyone `await`ing it will now block forever, because we'll never set the result of *that*.

Reply:
This class needs a bit of rethinking, I just iterated until the tests passed :) It's a bit hard to follow

Reply:
It's been slightly rethunk.